Move content to docs.0install.net

bastianeicher opened the pull request:
I have created a new static CMS-based documentation website for 0install: https://github.com/0install/docs

This combines documentation from http://0install.net/ and http://0install.de/. It is currently hosted at https://docs.0install.de/. I would like to move it to https://docs.0install.net/. Once this is done, this PR could be used to replace the existing documentation pages with redirects to their new equivalents.

bastianeicher commented:
The content is now available at https://docs.0install.net/.

talex5 commented:
Thanks - I'm working my way though it (have made a couple of PRs today). Some other quick things I noticed (I haven't read all of it yet):

- The copyright/license information from the footer has gone.
- There should probably be a link back to the main site. In particular, it's not obvious how to install it, once you've been convinced by the documentation...
- The ordering seems a bit strange. Why is "Trouble-shooting" before "Basics" for example? Perhaps Comparison/Features/Perspectives should be under "About"?

bastianeicher commented:
I've just added the following footer to https://docs.0install.net/, based on the one from http://0install.net/:
> © Copyright 2003-2019 Thomas Leonard and others
> Permission is granted to use the documentation in accordance with the terms of the Creative Commons Attribution-ShareAlike 2.5 license.

Good point about the link back to the main site. Where do you think would be a good place to put it? Perhaps the "Zero Install" title in the top-left (which currently just links to the documentation's main page)?

Currently the ordering is alphabetical by file name, with directories before files. Due to some clever naming and some dumb luck I managed to get the structure looking pretty decent that way.
mkdocs also supports explicit ordering of pages, however if we choose to do that, we'll need to list all of the pages explicitly. I really liked the fact that with the current setup, simply adding a new `.md` file is enough to add a new page.

talex5 commented:
Thanks. By the way, feel free to list yourself as a copyright owner in the footer too.

> Good point about the link back to the main site. Where do you think would be a good place to put it? Perhaps the "Zero Install" title in the top-left (which currently just links to the documentation's main page)?

That sounds good to me.

> Currently the ordering is alphabetical by file name, with directories before files.

This is a bit confusing with the tools because e.g. 0compile and 0release have pages listed near the start, but subdirectories with the same name at the end.

bastianeicher commented:
> Thanks. By the way, feel free to list yourself as a copyright owner in the footer too.

Done.

> > Good point about the link back to the main site. Where do you think would be a good place to put it? Perhaps the "Zero Install" title in the top-left (which currently just links to the documentation's main page)?
> 
> That sounds good to me.

Done.

> This is a bit confusing with the tools because e.g. 0compile and 0release have pages listed near the start, but subdirectories with the same name at the end.

I've moved the 0compile and 0release pages into their subdirectories for now. Still not a perfect solution, but a little better I think.

bastianeicher commented:
I made some changes to the menu structure:

- Renamed About to Home
- Moved Perspectives, Comparison, Features and FAQ under a new About section
- Moved Trouble-Shooting under Details

talex5 commented:
That looks better :-)

BTW, you have some links like `[OSTree](../https://ostree.readthedocs.io/)` now...

bastianeicher commented:
> BTW, you have some links like `[OSTree](../https://ostree.readthedocs.io/)` now...

Whoops, too hasty with search & replace... Fixed now.

talex5 commented on 0bootstrap.html:
```diff
@@ -0,0 +1 @@
+﻿<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd"><html lang="en"><head><meta http-equiv="REFRESH" content="0;url=https://docs.0install.net/tools/0bootstrap/"></head><body><a href="https://docdocs.0install.netols/0bootstrap/">try here</a></body>
```
Doesn't look right: `https://docdocs.0install.netols/0bootstrap/`

bastianeicher replied:
Oops. Looks like the `<meta>` redirects were pointing to the correct URLs but most of the `<a>`s were broken. Should all be fixed now.

bastianeicher committed:
Move content to docs.0install.net

talex5 commented:
Merged as #6.
